lib/parsers/yarn-lock-parse.ts

```diff
@@ -100,18 +102,25 @@ export class YarnLockParser implements LockfileParser {
     };
 
     const depKey = `${searchedDep.name}@${searchedDep.version}`;
-
+    // tslint:disable
     const dep = _.get(lockFile.object, depKey);
 
-    if (!dep) {
+    if (!dep && strictOutOfSync) {
```

Comment:
If you don't mind can we keep `if (!dep) {` condition and check `if (strictOutOfSync)` inside?

```
if (!dep) {
      if (strictOutOfSync) {
           throw new OutOfSyncError(searchedDep.name, 'yarn');
      }
      depSubTree.version = searchedDep.version;
      depSubTree.missingLockFileEntry = true;
      return depSubTree;
}
```

Reply:
Same for package-lock